src/utils/restClient/index.ts

```diff
@@ -0,0 +1,83 @@
+import fetch, { Response, RequestInit } from 'node-fetch';
+import { v4 as uuidv4 } from 'uuid';
+
+const DEFAULT_TIMEOUT = 10000;
+const DEFAULT_MAX_RETRIES = 2;
+const BASE_URL = 'https://api.mercadopago.com';
+
+interface RestClientConfig {
+  timeout?: number;
+  queryParams?: Record<string, string | number>;
+  retries?: number;
+  idempotencyKey?: string;
+}
+
+class RestClient {
+	private static generateIdempotencyKey(): string {
+		return uuidv4();
+	}
+
+	private static appendQueryParamsToUrl(url: string, queryParams?: Record<string, string | number>): string {
+		if (!queryParams) return url;
+
+		const searchParams = new URLSearchParams();
+
+		for (const key in queryParams) {
+			if (Object.prototype.hasOwnProperty.call(queryParams, key)) {
+				searchParams.append(key, queryParams[key].toString());
+			}
+		}
+
+		return url.includes('?') ? `${url}&${searchParams.toString()}` : `${url}?${searchParams.toString()}`;
+	}
+
+	static async fetch<T>(
+		endpoint: string,
+		config?: RestClientConfig & RequestInit
+	): Promise<T> {
+		const {
+			timeout = DEFAULT_TIMEOUT,
+			retries: maxRetries = DEFAULT_MAX_RETRIES,
```

Comment:
Em alguns sdks (dotnet e acho que java) a gente deixa o integrador não só configurar o número de tentativas como ele pode configurar a estratégia de retry que ele quer usar, é só ele implementar um interface. Acha que vale a pena fazer a mesma coisa no sdk de node?

Reply:
nesse caso eu poderia disponibilizar parametros de configuração, como o fator exponencial do retire, e até mesmo o delay entre cada retrie, vc acha que seria util? nesse momento deixei como fatore exponencial 2x e o delay como 1000

bastante baseado no [artigo](https://www.codewithyou.com/blog/how-to-implement-retry-with-exponential-backoff-in-nodejs) que vc me recomendou

Reply:
Acho que seria útil sim.